cdk/tests/unit/config/test_config.py

```diff
@@ -57,6 +57,28 @@ def test_invalid_version_syntax(self):
         with self.assertRaisesRegex(ValueError, f"Invalid version string: '{suffixed_schema}'"):
             config_loader(c)
 
+    def test_eks_ng_az_mismatch(self):
+        c = config_template().render()
+        c["vpc"]["max_azs"] = 3
+        c["eks"]["unmanaged_nodegroups"]["platform-0"]["availability_zones"] = [
+            "us-west-2a",
+            "us-west-2b",
+            "us-west-2c",
+        ]
+        config_loader(c)
+
+        c = config_template().render()
+        c["vpc"]["max_azs"] = 3
+        c["eks"]["unmanaged_nodegroups"]["platform-0"]["availability_zones"] = [
+            "us-west-2a",
+            "us-west-2d",
+            "us-west-2c",
+        ]
+        with self.assertRaisesRegex(
+            ValueError, "Nodegroup platform-0 has availability zone us-west-2d exceeding vpc.max_azs count: 3"
```

Comment:
the test says 3 AZs, the list has 3, so why does it fail?

Reply:
Other comment explains the reasoning, but it's the presence of `us-west-2d` (the "fourth" AZ) that causes the failure.

Reply:
so I dont see how d is the fourth. There are 3 defined AZs, a, d, c, now if it was a,d,c,X then I agree you have a forth. Asking for a max of 3 doesn't guarantee a,b,c.